ipn/ipnlocal/local.go

```diff
@@ -1811,6 +1811,21 @@ func (b *LocalBackend) readPoller() {
 	}
 }
 
+// HostinfoUpdated is called to recompute the Hostinfo and send
+// the new version to the control server.
+func (b *LocalBackend) HostinfoUpdated() {
```

Comment:
this name sounds like a accessor that returns a bool

make it sound imperative? `ResendHostinfoIfNeeded` maybe? Otherwise LGTM.

Wow, I forgot how horrible that `doSetHostinfoFilterServices` is. TODOs about in that call chain. (cc @andrew-d)

Reply:
done.

some of the hostinfo call chain looks like it wouldn't be too hard to clean up.